core/src/main/java/io/questdb/griffin/engine/join/LtJoinNoKeyFastRecordCursorFactory.java

```diff
@@ -0,0 +1,362 @@
+                slaveTimestamp = slaveRecA.getTimestamp(slaveTimestampIndex);
+                if (slaveTimestamp >= masterTimestamp) {
+                    return true;
+                }
+                record.hasSlave(true);
+                slaveCursor.recordAt(slaveRecB, Rows.toRowID(slaveFrameIndex, slaveFrameRow));
+                slaveFrameRow++;
+            }
+            return false;
+        }
+
+        private void nextSlave(long masterTimestamp) {
+            final TimeFrame frame = slaveCursor.getTimeFrame();
+            while (true) {
+                if (frame.isOpen() && frame.getIndex() == slaveFrameIndex) {
+                    // Scan a few rows to speed up self-join/identical tables cases.
+                    if (linearScan(frame, masterTimestamp)) {
+                        return;
+                    }
+                    if (slaveFrameRow < frame.getRowHi()) {
+                        // Fallback to binary search.
+                        long foundRow = binarySearch(masterTimestamp, slaveFrameRow, frame.getRowHi() - 1);
+                        if (foundRow < slaveFrameRow) {
+                            // All searched timestamps are equal or greater than the master timestamp.
+                            // Linear scan must have found the row.
+                            return;
+                        }
+                        slaveFrameRow = foundRow;
+                        record.hasSlave(true);
+                        slaveCursor.recordAt(slaveRecB, Rows.toRowID(slaveFrameIndex, slaveFrameRow));
+                        slaveTimestamp = slaveRecB.getTimestamp(slaveTimestampIndex);
+                        if (foundRow < frame.getRowHi() - 1 || slaveTimestamp == masterTimestamp - 1) {
+                            // We've found the row, so there is no point in checking the next partition.
+                            return;
+                        }
+                    }
+                }
+                if (!openSlaveFrame(frame, masterTimestamp)) {
+                    return;
+                }
+            }
+        }
+
+        private void of(RecordCursor masterCursor, TimeFrameRecordCursor slaveCursor) {
+            this.masterCursor = masterCursor;
+            this.slaveCursor = slaveCursor;
+            masterRecord = masterCursor.getRecord();
+            slaveRecA = slaveCursor.getRecord();
+            slaveRecB = slaveCursor.getRecordB();
+            record.of(masterRecord, slaveRecB);
+            slaveTimestamp = Long.MIN_VALUE;
+            slaveFrameRow = Long.MIN_VALUE;
+            slaveFrameIndex = -1;
+            toTop();
+        }
+
+        private boolean openSlaveFrame(TimeFrame frame, long masterTimestamp) {
+            while (true) {
+                if (isSlaveOpenPending) {
+                    if (slaveCursor.open() < 1) {
+                        // Empty frame, scan further.
+                        isSlaveOpenPending = false;
```

Comment:
nit: not covered by tests

Reply:
Fair point. Do you happen to know the easiest way to produce an empty partition?